doc/arkode/guide/source/Usage/LSRKStep/User_callable.rst

```diff
@@ -0,0 +1,394 @@
+.. ----------------------------------------------------------------
+   Programmer(s): Mustafa Aggul @ SMU
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2024, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _ARKODE.Usage.LSRKStep.UserCallable:
+
+LSRKStep User-callable functions
+==================================
+
+This section describes the LSRKStep-specific functions that may be called
+by the user to setup and then solve an IVP using the LSRKStep time-stepping
+module.  As mentioned in Section :numref:`ARKODE.Usage.UserCallable`,
+shared ARKODE-level routines may be used for the large majority of LSRKStep
+configuration and use.  In this section, we describe only those routines
+that are specific to LSRKStep.
+
+As discussed in the main :ref:`ARKODE user-callable function introduction
+<ARKODE.Usage.UserCallable>`, each of ARKODE's time-stepping modules
+clarifies the categories of user-callable functions that it supports.
+LSRKStep supports the following categories:
+
+* temporal adaptivity
+
+
+
+.. _ARKODE.Usage.LSRKStep.Initialization:
+
+LSRKStep initialization functions
+---------------------------------
+
+
+.. c:function:: void* LSRKStepCreateSTS(ARKRhsFn rhs, sunrealtype t0, N_Vector y0, SUNContext sunctx);
+
+   This function allocates and initializes memory for a problem to
+   be solved using the LSRKStep time-stepping module in ARKODE.
+
+   **Arguments:**
+      * *rhs* -- the name of the C function (of type :c:func:`ARKRhsFn()`)
+        defining the right-hand side function.
+      * *t0* -- the initial value of :math:`t`.
+      * *y0* -- the initial condition vector :math:`y(t_0)`.
+      * *sunctx* -- the :c:type:`SUNContext` object (see :numref:`SUNDIALS.SUNContext`)
+
+   **Return value:**
+      If successful, a pointer to initialized problem memory
+      of type ``void*``, to be passed to all user-facing LSRKStep routines
+      listed below.  If unsuccessful, a ``NULL`` pointer will be
+      returned, and an error message will be printed to ``stderr``.
```

Comment:
Yes.  The behavior would depend on how `arkProcessError` works (which would depend on the error handler and build options).  It also looks like some of the steppers could `return(NULL)` without calling `arkProcessError` (e.g., ARKStep when `arkAllocVec` fails).  It would be good to have a separate PR that ensures consistent behavior, and that updates the user guides as necessary.